pkg/controller/direct/gkehub/featuremembership_controller.go

```diff
@@ -121,6 +121,13 @@ func (a *gkeHubAdapter) Find(ctx context.Context) (bool, error) {
 
 // Delete implements the Adapter interface.
 func (a *gkeHubAdapter) Delete(ctx context.Context) (bool, error) {
+	exist, err := a.Find(ctx)
+	if err != nil {
+		return false, fmt.Errorf("finding feature for %s:%w", a.featureID, err)
+	}
+	if !exist {
+		return false, fmt.Errorf("expecting feature for %s, but not found", a.featureID)
+	}
```

Comment:
sanity check: don't we want to support `exists == false` as it doesn't exist. I'm thinking what if end up reconciling a stale `delete` event  for instance, we will err out and fail the reconcile possibly?

https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/aaa5bbdaea628288685cded6b54166d6fda795cf/pkg/controller/direct/directbase/directbase_controller.go#L241-L255

It may be sufficient to just error out not found.

Reply:
Do you have a scenario in your mind that KCC controller may reconcile a stale delete event?

I am leaning towards to error out on unexpected delete reconciliations.

Reply:
Thanks for explaining, returning (false, nil) then if notfound

Reply:
Sweet, yes, not returning err on not found was what I wanted to make sure we do 😛 

Thanks Justin for clarifying and Ziyue for following up!